### PR TITLE
Display commit SHA in build table

### PR DESCRIPTION
**Pull request by el-mapache**
* Adds link to the commit

**Screenshot**
screen shot 2017-12-01 at 9 08 38 am

### CODE REVIEW

**Comment by el-mapache** [comment]
I modified the `GitHubRepoLink` component a bit, let me know if y'all would rather see it split up into different components, e.g. an external link component that the github link component could import

**Comment by jseppi** [comment]
I do think a separate component(s) might be a good idea at this point. Could you whip that up?

**Comment by el-mapache** [comment]
yes, definitely!

**Comment by jseppi** [comment]
@el-mapache: you should be able to locally run `yarn lint:diff` to see the same linting errors that CodeClimate will find

**Comment by el-mapache** [comment]
@jseppi I did, but it seemed to only show two errors. Trying again!

**Comment by el-mapache** [comment]
@jseppi All fixed up!

File: test/frontend/components/site/SiteGitHubBranches.test.jsx [hunk 1]
```diff
@@ -29,15 +29,24 @@ describe('<SiteGitHubBranches />', () => {
...
     const rows = wrapper.find('tbody tr');
+
     expect(rows).to.have.length(2);
     expect(rows.at(0).find('td').at(0).text())
       .to.have.string('branch-one');
     expect(rows.at(1).find('td').at(0).text())
       .to.have.string('branch-two');
-    expect(rows.find('GitHubRepoLink')).to.have.length(2);
+
+    // TODO: Nested component selectors, like the `Connect(BranchViewLink)` below, don't
```
Comment (jseppi) [inline test/frontend/components/site/SiteGitHubBranches.test.jsx]: Let's file an issue for this and reference it here in the comment

Comment (el-mapache) [reply test/frontend/components/site/SiteGitHubBranches.test.jsx]: sure thing

**Comment by line47** [comment]
maybe "view commit" for the text would be good with `title` text containing the commit value that way we don't have commit numbers breaking the table

**Comment by el-mapache** [comment]
@jseppi I Opened an issue here https://github.com/18F/federalist/issues/1325, take a look and see if it seems reasonable. I already pushed up a commit linking to it in the code.

**Comment by el-mapache** [comment]
@line47 That sounds good to me!

**Comment by jseppi** [comment]
@el-mapache: issue looks good to me, thanks!

**Comment by el-mapache** [comment]
Updated code based on @line47's suggestion!
screen shot 2017-12-04 at 5 25 54 pm

**Comment by wslack** [comment]
Why don't we put the "view commit" link in parenthesis next to the branch name?

Something like this but with a carriage return so the link is always below the branch name:
screen shot 2017-12-04 at 5 33 38 pm

**Comment by el-mapache** [comment]
@wslack Seems fine, the majority of users probably don't need to know what the SHA is, just that they can get back to the commit. @line47 what do you think?

**Comment by wslack** [comment]
The most compact way to do it would be to turn the branch name into a link to the commit but it wouldn't be obvious to users that the links pointed differently.

**Comment by line47** [comment]
@el-mapache yeah I think that sounds OK
@wslack we definitely want to make sure the linking experience is consistent throughout the application UI to avoid confusion.

**Comment by el-mapache** [comment]
@line47 then, should we be using the :octocat: icon for these links as well? The other external github links don't have the links on a separate line, they have a line of text with the :octocat: serving as the link.

**Comment by wslack** [comment]
@line47 agreed but these are different than links to the repo in their behavior and so I think (View Commit) without a GitHub icon is fine.

**Comment by line47** [comment]
I feel the GitHub icon used in conjunction with a link would be most clear approach and avoid confusion on what _is_ and _is not_ a link to GitHub.

**Good example**
[GitHub link](www.federalist.18f.gov) :octocat: 
In this example the link and icon are both inside the same `<a>` tag

**Confusing example**
GitHub link [:octocat:](www.federalist.18f.gov)
In this example the text is not a link and the icon is a link.

Either direction we go let's be consistent and either use GitHub icons in conjunction with a link or don't use Github icons at all and just use link text. I think the icon with text is more clear you will be taken to GitHub.

**Comment by el-mapache** [comment]
@line47 I agree with that, I think that having the icon be the actual link could potentially be confusing to users who don't know much about github, or might not realize that you can click it.

Either the plain link or the 'good' example above seems like a good direction!

**Comment by line47** [comment]
Also wanted to point out this issue which seems to co-inside with this PR. https://github.com/18F/federalist/issues/1266 

@el-mapache if you had some time to pair on making those changes in that PR that would be awesome as I'd like some extra eyes on making those changes.

**Comment by el-mapache** [comment]
@line47 yes, definitely!

**Comment by el-mapache** [comment]
@wslack @jseppi @line47 What do yall think about merging this PR for the functionality? I can put in a separate PR for normalize the link styles (or we can roll it into the PR Scott linked above).

**Comment by line47** [comment]
I vote for merge and work and open a new PR 🎉

**Comment by wslack** [comment]
I would only want us to merge this into staging if we can get the change done before Friday. I don't think I want us to push a new column to the builds table live - instead, this should be a plain link, with no GitHub icon.

Right now, I honestly feel like having a prose link and a Github icon just takes up extra space.

**Comment by jseppi** [comment]
Going to update this branch with `staging` now that #1323 has been merged in. Hopefully it'll kick CodeClimate into getting unstuck.

**Comment by el-mapache** [comment]
Here is the updated 'view commit' link
screen shot 2017-12-06 at 12 49 35 pm

**Comment by el-mapache** [comment]
@line47 @jseppi @wslack commit spam!

**Comment by wslack** [comment]
I'm good with this but someone else should review the code updates

**Pull request merged**